### ISSUE TITLE
[BUG] Snoozed tabs disappearing (not showing in snooze list)

### ISSUE THREAD

**Issue by Oshyan** [rohanb10/snoozz-tab-snoozing#84]
## Description
I don't know exactly what's happening or when it started, but I'm seeing something very concerning. Some time within the last few days I noticed that tabs I thought I had snoozed were not showing in the snoozed tabs list and would not show up in searches. This includes both single tabs and groups of tabs (I'm aware of the issue with searching in groups). It appears that it has happened from the regular action of clicking the Snoozz extension button. The previously woken-up items for the "lost" tabs show up in history, until...

This came to a head just now when, after identifying a tab that was no longer set to wake when I intended, I attempted to "re-snooze" it from my History. I had searched for its name, found it in previously woken tabs, then used the re-snooze button to attempt to snooze it for next Monday, the 3rd of January. Here's where it got really concerning. It disappeared entirely when I did that. I had the search term still entered, but now the entry that had shown from the history disappeared. Clearing the search term, I see that **it is now not found in future tabs waking up NOR in history**! That's deeply concerning because it means I might not even have a way of retrieving tabs that "mis-snoozed" when this is fixed.

 In fact there are *no* tabs showing to awaken on the 3rd, and I'm almost certain I had others set for that day. In fact between Friday the 31st of this week, and Thursday the 6th of next week, *no tabs show to awaken*, and I'm quite certain a number of them should be scheduled in there. Hopefully they're now at least in my history...

I rely extremely heavily on this kind of extension. I have just over 100 tabs currently snoozed. And now I don't know which ones may be randomly not snoozing, or how long this has been happening, or whether I'll be able to piece together the ones that were missed.

## Expected Behavior
Tabs should be set to awaken when specified
Snoozing tabs should show up correctly in the list of all snoozed tabs
Snoozing tabs should be searchable with matching key words

## Actual Behavior
Some tabs are not showing up at all. It's possible they are just hidden due to some bug and actually will awaken. I will not find out until the 3rd, I suppose. Some tabs for Tomorrow (December 30th) and Friday (31st) still show. 

## How to Reproduce (if applicable)
I'm not certain except to just try to snooze a tab. Or perhaps the most significant bug to reproduce would be the specific case of attempting to "re-snooze" from the History section of the Sleeping Tabs list.

## Your Device
* Browser and version: Google Chrome 96.0.4664.110
* OS: Windows 10 Pro
* Device Resolution (UI related bugs only)

## Which Page
Nap Room/Sleeping Tabs List

**Comment by dmurat** [comment]
My guess is that the bug is somehow related to the year switch. In my case snoozing with Saturday, Next Monday, and Next Week buttons cause tabs to disappear.  Other snoozing buttons work correctly, as it seems. Will see the situation after the new year.

**Comment by Oshyan** [comment]
Yeah, that's kind of what I was thinking too. Related to the year change. But it's odd that it's just certain days and not even a full week in the new year (I think).

**Comment by polanar** [comment]
I'm seeing the same thing. I know I have many tabs snoozed to open over the next few days - including some which were snoozed today (31/12) for tomorrow morning and evening, but the first ones showing up in the Nap Room are for 9th January.

I am very concerned that these snoozed tabs have been lost.

**Comment by Oshyan** [comment]
Well, I had a number of tabs wake up today that were not listed in Snoozed Tabs. *And* I now see what appears to be a normal amount of tabs scheduled for the days in the coming week, which were previously invisible up to January 6th. So it *appears* they have not been lost, just some kind of display/list bug that was related to the year change perhaps. 😌 Can others who were seeing this confirm?

**Comment by polanar** [comment]
Same here - the tabs that disappeared seem to have reappeared. However I can't be certain that none were lost - there are a handful which I snoozed yesterday (or edited the wakeup time for) which don't seem to be in the list now.

**Comment by dmurat** [comment]
All tabs reappeared for me.

**Comment by Cologler** [comment]
Same here. 

https://github.com/rohanb10/snoozz-tab-snoozing/blob/632d067c80b901136251b6f2540378b20c64c630/scripts/nap_room.js#L344-L360

`getTimeGroup` may return `undefined` if not match any time group.

**Comment by codeLord61** [comment]
I am having the same issue. It is December 31 of 2024 at the time of writing and I have snoozed a window full of tabs that is supposed to wake up on January 1 of 2025. I tested @dmurat's theory, snoozed a tab that will wake up after an hour within 31st and that appears on the snooze list so yeah its a next year issue.

**Comment by polanar** [comment]
Same issue this year - on 31 December, most (but not all) of the snoozed tabs for the next week to ten days disappeared. And this morning, 1 January, they seem to be back. I can't be certain that no tabs were lost, but this does seem to be more of a display issue rather than a lost-snoozes issue.

**Comment by Oshyan** [comment]
That at least is somewhat comforting to hear! I didn't think to check it again this year, but I'm glad there is still discussion around it even if there is no development.
